src/servers/http_server_v2.cc

```diff
@@ -846,6 +868,97 @@ HTTPAPIServerV2::HandleServerHealth(
   TRTSERVER_ErrorDelete(err);
 }
 
+void
+HTTPAPIServerV2::HandleRepositoryIndex(
+    evhtp_request_t* req, const std::string& repository_name)
+{
+  if (req->method != htp_method_GET) {
+    evhtp_send_reply(req, EVHTP_RES_METHNALLOWED);
+    return;
+  }
+
+  TRTSERVER_Error* err = nullptr;
+  const char* const* models;
+  uint64_t models_count;
+  TRTSERVER2_ModelIndex* model_indices = nullptr;
+  if (repository_name.empty()) {
+    err = TRTSERVER2_ServerModelIndex(server_.get(), &model_indices);
+    if (err == nullptr) {
+      err = TRTSERVER2_ModelIndexNames(model_indices, &models, &models_count);
+    }
+  } else {
+    err = TRTSERVER_ErrorNew(
+        TRTSERVER_ERROR_UNSUPPORTED,
+        "'repository_name' specification is not yet supported");
+  }
+
+  rapidjson::Document document;
+  document.SetObject();
+  rapidjson::Document::AllocatorType& allocator = document.GetAllocator();
+  if (err == nullptr) {
+    rapidjson::Value models_array(rapidjson::kArrayType);
+    LOG_VERBOSE(1) << "models size: " << models_count;
+    for (uint64_t i = 0; i < models_count; i++) {
+      rapidjson::Value model_index;
+      model_index.SetObject();
+      std::string model_str(models[i]);
```

Comment:
It is a little strange that you convert to a std::string and then extract the c_str out of that. Why not just use model directly and strlen(model)?